vite.config.ts

```diff
@@ -0,0 +1,103 @@
+import path from "path";
+import react from "@vitejs/plugin-react";
+import browserslistToEsbuild from "browserslist-to-esbuild";
+import { webpackStats } from "rollup-plugin-webpack-stats";
+import { defineConfig } from "vite";
+import { VitePWA } from "vite-plugin-pwa";
+
+export default defineConfig({
+  root: "./",
+  publicDir: "./server/static",
+  server: {
+    port: 3001,
+  },
+  plugins: [
+    // https://github.com/vitejs/vite-plugin-react/tree/main/packages/plugin-react#readme
+    react({
+      babel: {
+        babelrc: true,
+        presets: [
+          "@babel/preset-react",
+          "@babel/preset-typescript",
+          [
+            "@babel/preset-env",
+            {
+              corejs: {
+                version: "3",
+                proposals: true,
+              },
+              useBuiltIns: "usage",
+              exclude: [
+                "transform-modules-commonjs",
+                "proposal-dynamic-import",
+              ],
+            },
+          ],
+        ],
+      },
```

Comment:
To exclude the two plugins from `@babel/preset-env` I’ve had to copy the whole `presets` part of the babel config, but it’s using the existing babel config for the rest.

It’s not possible to just add this to the babel config, because it’s used for the backend and the frontend.